openc3/python/openc3/io/serial_driver.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2025 OpenC3, Inc.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# This file may also be used under the terms of a commercial license
+# if purchased from OpenC3, Inc.
+
+import time
+import threading
+from typing import Optional, Union
+import serial
+
+class SerialDriver:
+    """A platform independent serial driver"""
+    
+    # Parity constants
+    EVEN = 'EVEN'
+    ODD = 'ODD'
+    NONE = 'NONE'
+    VALID_PARITY = [EVEN, ODD, NONE]
+    
+    def __init__(self,
+                 port_name: str,
+                 baud_rate: int,
+                 parity: str = 'NONE',
+                 stop_bits: int = 1,
+                 write_timeout: float = 10.0,
+                 read_timeout: Optional[float] = None,
+                 flow_control: str = 'NONE',
+                 data_bits: int = 8,
+                 read_polling_period: float = 0.01,
+                 read_max_length: int = 1000):
```

Comment:
1000 seems low for the read max length. @ryanmelt I know this is the same as the Ruby code but where did this come from?